cachi2/core/models/sbom.py

```diff
@@ -21,6 +21,13 @@ class Property(pydantic.BaseModel):
     value: str
 
 
+class ExternalReference(pydantic.BaseModel):
+    """An ExternalReference inside an SBOM component."""
+
+    url: str
+    type: Literal["distribution"] = "distribution"
+
+
```

Comment:
Can you justify this change? The commit message is empty with such an IMO substantial SBOM update. Why is this needed? Why have these specific fields been chosen for the external reference? Can you elaborate?

Reply:
It is sort of described in the follow-up commit message, as I'd say the introduction itself bears no meaning, if not used within the context of the generic fetcher (other package managers do not utilize it).

> Generic fetcher now generates SBOM components of purl type
pkg:generic. It also produces externalReferences of type
distribution for each generic component to allow for gating
at a later time.

I chose `externalReferences` of type `distribution` to describe the `download_url` in a way that's easily processable in whatever will look at the SBOM later (thinking about EC, but could be anything, really). The [CycloneDX spec](https://github.com/CycloneDX/guides/blob/main/SBOM/en/0x51-External-References.md#external-references) describes this type of externalReferences as "Direct or repository download location" which in my opinion makes sense for the `download_url`

Reply:
I think what you wrote here can be worked into a message for this specific commit. For now, it'll only be used for `generic`, but that might change in the future. The reasoning here is that this field should carry the URL from which a component was downloaded, as a means of having more meaningful data in the SBOM so that other tooling can build on top of it.

Reply:
I will amend the commit message for that commit.

Reply:
Amended.

Reply:
@kosciCZ  please include https://cyclonedx.org/docs/1.6/json/#components_items_externalReferences as a reference in the commit message where the following is IMO the most important bit justifying the selection of this particular component item:

> External references provide a way to document systems, sites, and information that may be relevant but are not included with the BOM. They may also establish specific relationships within or external to the BOM.

Reply:
Added to the commit message.